middleware/common/include/common/flag.h

```diff
@@ -6,176 +6,168 @@
       {
-         using enum_type = E;
-         using underlying_type = std::underlying_type_t< enum_type>;
-
-         static_assert( std::is_enum< enum_type>::value, "E has to be of enum type");
+         template< typename T>
+         concept has_description = enum_flag_like< T> && requires( T e)
+         {
+            { description( e)} -> std::same_as< std::string_view>;
+         };
          
+      } // detail
+      
+   } // common::flag
+   
+} // casual
 
-         //! just a helper to easier deduce the enum-type
-         constexpr static auto type() noexcept { return enum_type{};}         
-
-         constexpr Flags() = default;
-
-         template< typename... Enums>
-         constexpr Flags( enum_type e, Enums... enums) noexcept : Flags( bitmask( e, enums...)) {}
- 
-         constexpr explicit Flags( underlying_type flags) noexcept : m_flags( flags) {}
-
-         constexpr Flags convert( underlying_type flags) const
-         {
-            if( flags & ~underlying())
-               code::raise::error( code::casual::invalid_argument, "flags: ", flags, " limit: ", *this);
+template< casual::common::flag::enum_flag_like T>
+constexpr auto operator ! ( T value)
+{
+   return std::to_underlying( value) == 0;
+}
 
-            return Flags{ flags};
-         }
+template< casual::common::flag::enum_flag_like T>
+constexpr auto operator | ( T lhs, T rhs)
+{
+   return static_cast< T>( std::to_underlying( lhs) | std::to_underlying( rhs));
+}
 
-         template< typename E2>
-         constexpr Flags convert( Flags< E2> flags) const
-         {
-            return convert( flags.underlying());
-         }
+template< casual::common::flag::enum_flag_like T>
+constexpr auto& operator |= ( T& lhs, T rhs)
+{
+   return lhs = static_cast< T>( std::to_underlying( lhs) | std::to_underlying( rhs));
+}
 
-         constexpr bool empty() const noexcept { return m_flags == underlying_type{};}
+template< casual::common::flag::enum_flag_like T>
+constexpr auto operator & ( T lhs, T rhs)
+{
+   return static_cast< T>( std::to_underlying( lhs) & std::to_underlying( rhs));
+}
 
-         constexpr explicit operator bool() const noexcept { return ! empty();}
+template< casual::common::flag::enum_flag_like T>
+constexpr auto& operator -= ( T& lhs, T rhs)
+{
+   return lhs = static_cast< T>( std::to_underlying( lhs) & ~std::to_underlying( rhs));
+}
 
-         constexpr bool exist( enum_type flag) const noexcept
-         {
-            return ( m_flags & std::to_underlying( flag)) == std::to_underlying( flag);
-         }
 
-         platform::size::type bits() const noexcept
+namespace casual
+{
+   namespace common::flag
+   {
+      namespace native
+      {
+         template< auto flag>
+         constexpr bool exists( auto flags)
          {
-            return std::bitset< std::numeric_limits< underlying_type>::digits>{ static_cast< unsigned long long>( m_flags)}.count();
+            return ( flags & flag) == flag;
          }
```

Comment:
What is the intention/use of this? Casual builds  and the test suite executes without "common::flag::native::exists( auto flags)". Usage seems a bit odd to me. Seems to be a function template with a value template parameter?

Reply:
This to be able to check "native" flags.  `flag::native::exist< TPNOTRAN>( some_ulong)`.  Replacement for `has::flag< FLAG>( value)` 
I apparently got rid of all the uses with untyped flag-checking, we might as well remove this.